OpenRA.Mods.Common/Traits/Conditions/GrantConditionOnDeploy.cs

```diff
@@ -157,6 +160,17 @@ Activity IWrapMove.WrapMove(Activity moveInner)
 			return activity;
 		}
 
+		bool IDelayCarryallPickup.TryLockForPickup(Actor self, Actor carrier)
+		{
+			if (!Info.UndeployOnPickup || deployState == DeployState.Undeployed)
+				return true;
+
+			if (deployState == DeployState.Deployed)
```

Comment:
Adding `&& !IsTraitPaused && !IsTraitDisabled` here will make sure that EMPed units won't unemploy.

Reply:
Disabling a trait is supposed to behave equivalently to the trait not being defined at all, so the disabled check belongs on the `return true` case above.

Reply:
That does hint at a bit of a conceptual flaw though. If the unit is unable to be carried in its current deploy state, then losing the ability to transition between states doesn't make it suddenly okay to be carried while deployed. The correct behaviour when semi-permanently losing the ability to undeploy when `UndeployOnPickup == true` would be to refuse the pick-up altogether, i.e. we should return `LockResponse.Failed` in that case.

Reply:
The disabled filtering is actually already handled in the `Carryall.LockForPickup` so it doesn't matter what we are doing here in any case.